src/Infrastructure.Dapper/Repositories/UserRepository.cs

```diff
@@ -164,7 +185,52 @@ public async Task<IEnumerable<User>> GetManyAsync(IEnumerable<Guid> ids)
                 new { Ids = ids.ToGuidIdArrayTVP() },
                 commandType: CommandType.StoredProcedure);
 
+            UnprotectData(results);
             return results.ToList();
         }
     }
+
+    private void ProtectData(User user)
+    {
+        if (user == null)
+        {
+            return;
+        }
+
+        user.MasterPassword = user.MasterPassword == null ? null :
+            string.Concat("P_", _dataProtector.Protect(user.MasterPassword));
+        user.Key = user.Key == null ? null :
+            string.Concat("P_", _dataProtector.Protect(user.Key));
+    }
+
+    private void UnprotectData(User user)
+    {
+        if (user == null)
+        {
+            return;
+        }
+
+        if (user.MasterPassword?.StartsWith("P_") ?? false)
```

Comment:
There's **no** chance the protection could somehow emit this right?

Reply:
Let me investigate. Looks like it could.

Reply:
Logic has changed now, so I dont believe this is necessary.